### ISSUE TITLE
Odd Legend Rendering for Sequence Diagrams

### ISSUE THREAD

**Issue by chriskn** [plantuml-stdlib/C4-PlantUML#330]
Hi, 

I have the following issue: 

When I take your [sequence diagram example](https://github.com/plantuml-stdlib/C4-PlantUML/blob/master/samples/C4_Sequence%20Diagram%20Sample%20-%20bigbankplc.puml
) and rendered it locally on my Windows machine with version 1.2023.12 (or 1.2023.11) of PlantUML and GraphViz 9.0.0, I get the following result: 

![grafik](https://github.com/plantuml-stdlib/C4-PlantUML/assets/1077869/ea3101c1-85d9-4c82-8d92-79a0f01ccac3)

As you can see, the legend is rendered odd. Somehow, the background rectangles are moved up and overlap.  
I can reproduce this issue with all my sequence diagrams, independent of their content or legend type.

![grafik](https://github.com/plantuml-stdlib/C4-PlantUML/assets/1077869/f3d1bf63-0ce8-49b5-b488-c2f27e3c3525)
 
Thanks in advance

**Comment by kirchsth** [comment]
Hi @chriskn,

I can reproduce it even with a "simpler" plantuml script. The problem is the "teoz" layout.
If I deactivate it with an additional `!pragma teoz false` line the it looks ok.

E.g. instead of only 
```planttuml
!include <C4/C4_Sequence>
```
please use
```planttuml
!include <C4/C4_Sequence>
!pragma teoz false
```

Can you please try it on your side until we get a bugfix from PlantUML team.
The disadvantage of this workaround is that it cannot display nested boxes.

BR Helmut

```
@startuml

!pragma teoz true

' if you remove the comment of following line it looks ok
' !pragma teoz false

c1 -> c2 : call

legend right
<#transparent,#transparent>|<color:#000000>**Legend**</color> |
|<#438DD5><color:#3C7FC0> <U+25AF></color> <color:#FFFFFF> container <size:10></size></color> |
|<#85BBF0><color:#78A8D8> <U+25AF></color> <color:#000000> component <size:10></size></color> |
|<#transparent><color:#444444> <U+25AF></color> <color:#444444> container boundary <size:10>(dashed)</size></color> |
endlegend
@enduml
```

**Comment by kirchsth** [comment]
entered a bug in [plantuml forum](https://forum.plantuml.net/18480/teoz-produces-odd-legend-for-sequence-diagrams)

**Comment by chriskn** [comment]
Thank you @kirchsth, I can confirm that `!pragma teoz false` fixes the issue but hides nested groups

**Comment by kirchsth** [comment]
Hi @chriskn,

we get a PlantUML fix in v1.2024.0beta3, you can download from https://github.com/plantuml/plantuml/releases/tag/snapshot

Can you please test it with your diagrams

Thank you and best regards
Helmut

**Comment by chriskn** [comment]
Hi @kirchsth, the bug is fixed with [v1.2024.1](https://github.com/plantuml/plantuml/releases/tag/v1.2024.1)

Thank you!
